library/src/main/java/com/macroyau/thingspeakandroid/ThingSpeakChannel.java

```diff
@@ -113,16 +113,38 @@
      * @param channelId The ID of this specific Channel.
      */
     public ThingSpeakChannel(long channelId) {
-        this(channelId, null);
+        this(channelId, null, null);
     }
 
     /***
-     * Constructor for private Channels.
+     * Constructor for public Channels.
+     *
+     * @param channelId The ID of this specific Channel.
+     * @param API The address of the custom thingspeak server API.
+     */
+    public ThingSpeakChannel(long channelId, URL API) {
```

Comment:
Use `String` instead of the `URL` class. Also, variable names should be in camel case (i.e., `api`).